src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import ChannelRow from "../ChannelRow/ChannelRow";
+import VideoRow from "../VideoRow/VideoRow";
+import "./SearchBar.css";
+
+const SearchBar = ({ inputSearch }) => {
+  const [channelRow, setChannelRow] = useState({});
+  const [videoRows, setVideoRows] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setChannelRow("");
+    setVideoRows([]);
+
+    const fetchChannelData = async () => {
+      try {
+        const response = await fetch(
+          `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=1&type=channel&q=${inputSearch}&safeSearch=none&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`
+        );
+        const data = await response.json();
+
+        createChannelRow(data["items"][0]);
+      } catch (error) {
+        alert("Search not found");
+        setIsError(true);
+        setIsLoading(false);
+      }
+    };
+
+    fetchChannelData();
+
+    const fetchVideosData = async () => {
+      try {
+        const response = await fetch(
+          `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=9&type=video&q=${inputSearch}&safeSearch=none&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`
+        );
+        const data = await response.json();
+
+        createVideoRows(data["items"]);
+        setIsError(false);
+      } catch (error) {
+        alert("Search not found");
+        setIsError(true);
+        setIsLoading(false);
+      }
+    };
+
+    fetchVideosData();
```

Comment:
you are calling `fetchVideosData()` after `fetchChannelData()`. Suppose if there is some error occured in `fetchChannelData()`, still your code will call `fetchVideosData()` so is this case considered?